New Resource: `azurerm_storage_discovery_workspace` (#31479)
terraform-providers/terraform-provider-azurerm: website/docs/r/storage_discovery_workspace.html.markdown

  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  description         = "Example Storage Discovery Workspace"
  sku                 = "Standard"

  workspace_root = [data.azurerm_subscription.current.id]

  scope {
    display_name   = "Production Storage"
    resource_types = ["Microsoft.Storage/storageAccounts"]
    tag_keys_only  = ["environment", "department"]
    tags = {
      criticality = "high"
    }
  }

  tags = {
    environment = "production"
  }
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) Specifies the name of the Discovery workspace. Must be 4-64 characters long, start with a letter, and contain only letters, numbers, and hyphens (no consecutive hyphens). Changing this forces a new resource to be created.

* `resource_group_name` - (Required) Specifies the name of the Resource Group where the Storage Discovery Workspace should exist. Changing this forces a new resource to be created.

* `location` - (Required) Specifies the Azure Region where the Discovery resource is created. Changing this forces a new resource to be created.

> * `scope` - (Required) One or more `scope` blocks as defined below. Maximum of 10 scopes. Changing this forces a new resource to be created (the Azure API does not allow editing scopes after the workspace exists).

Reviewer: "does not allow editing scope after.." wording is redundant with "Changing this forces a new resource to be created". Also implementation detail has to be moved to a note section as per reference-documentation-standards.md

```suggestion
* `scope` - (Required) One or more `scope` blocks as defined below. Changing this forces a new resource to be created.

-> **Note:** A maximum of 10 scopes are supported.
```

Please apply the same concern to other parts of this doc.

* `workspace_root` - (Required) A set of top-level Azure resource identifiers (Subscription IDs or Resource Group IDs) where Storage Discovery initiates its scan for storage accounts. You cannot specify both a subscription and its child resource group. Maximum of 100 items.

-> **Note:** The user or service principal must have at least Reader access (`Microsoft.Storage/storageAccounts/read`) on each specified root. The default limit of 100 can be increased by contacting Azure Support.

> * `description` - (Optional) A description for the Discovery workspace resource.

Reviewer: Please document the max 1024 character validation constraint.

* `sku` - (Optional) Specifies the Storage Discovery pricing plan. Possible values are `Free` and `Standard`. Defaults to `Standard`. See [Understand Storage Discovery Pricing](https://learn.microsoft.com/azure/storage-discovery/understand-pricing) for details.

* `tags` - (Optional) A mapping of tags which should be assigned to the resource.

---

A `scope` block supports the following:

* `display_name` - (Required) The display name for this scope. Must be 4-64 characters long and can only contain letters, numbers, spaces, and hyphens. Cannot start or end with a number, space, or hyphen, and cannot contain consecutive hyphens or spaces.

> * `resource_types` - (Required) A set of Azure resource type strings to include in this scope. Possible values are `Microsoft.Storage/storageAccounts`. Duplicate values are not allowed; order is not significant.

Reviewer: The sentence `Duplicate values are not allowed; order is not significant.` is redundant because at the beginning we already specified this attribute is a set.

```suggestion
* `resource_types` - (Required) A set of Azure resource type strings to include in this scope. Possible values are `Microsoft.Storage/storageAccounts`.
```

Author: Updated 👍
